manifests/profile/kubernetes/haproxy.pp

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# Kubernetes HAProxy profile
+#
+# This is for the gateway to a kubernetes cluster. In kubernetes terms,
+# this exists outside the actual cluster, but, since we need to be able
+# to access the cluster, this role is essential.
+#
+# @param master Exactly one gateway per cluster should be designated as
+#   the master so that any others will simply exist as failover nodes.
+# @param cluster The name of the cluster to serve as ambassador to. This
+#   defaults to nebula::profile::kubernetes::cluster, and you should
+#   almost certainly set that instead.
+class nebula::profile::kubernetes::haproxy (
```

Comment:
It might be worth looking into the puppetlabs haproxy module, but this is certainly OK for now.

Reply:
Could do, but the long-term plan is to have these manage their own configs, rather than have puppet manage them. Once puppet isn't doing that anymore, I doubt there'd be much an HAProxy module can do for us.